### ISSUE TITLE
[Issue] Environment commands don't seem to work correctly

### ISSUE THREAD

**Issue by IEvangelist** [Azure/azure-dev#2104]
- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
Run `azd version` and copy and paste the output here:

```
azd version 0.8.0-beta.3-pr.2738048 (commit 4230bfcb69f0426e612ba95a8da29c5318552af8)
```

**Describe the bug**
Description of issue you're seeing...

When calling `azd env list`, I see the two expected environments. But when I call `azd up`, it thinks I'm in an old environment that no longer exists.

**To Reproduce**
Steps to reproduce the behavior...

1. Create multiple environments.
2. Manually delete an environment.
3. Check environments and ensure you have a `default`.
4. Call `azd up`.

**Expected behavior**

Only environments that are listed in the `.azure/` folder should be valid environments.

**Environment**
Information on your environment:
    * Language name and version
    * IDE and version : [e.g. Visual Studio 16.3]

**Additional context**

![image](https://user-images.githubusercontent.com/7679720/236550883-51e70492-9aba-43ac-80bc-4a4875cf1523.png)

**Comment by vhvb1989** [comment]
@IEvangelist azd doesn't expect manual environments deletion.
The default environment is defined in `config.json`

![image](https://user-images.githubusercontent.com/24213737/236564405-020bdc1e-98f0-4c29-906c-6f3c15fff918.png)

If you remove the env folder inside .azure but you don't update the `config.json`, the state becomes `un-expected behavior` 

pd. I love your deftones background :)

**Comment by IEvangelist** [comment]
In my _.azure/_ folder, there are only two configs, and neither of them is the `dapine-07`, and the `default` defined in the _config.json_ is the value represented by the output of the call to:

```
azd env list

NAME        DEFAULT
davidpine   false
davidpine7  true
```

Any ideas?

**Comment by ellismg** [comment]
I spent a bunch of time trying to reproduce this issue but was unable to - and we've made a bunch of fixes since 0.8.0 and today as part of our bug bash during the lead up to 1.0.0, so I'm suspecting this is no longer an issue.

The only other possible thing I could think of would be that `AZURE_ENV_NAME` would be defined in your OS Environment, and in that case `azd` would be respecting that value.  If it pointed to a name that didn't exist, like `dapine-07` then you might see this behavior.

If you're still seeing this on 1.0.2, feel free to re-open this and we can dig in.
